mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala

```diff
@@ -154,13 +155,19 @@ class CountVectorizer @Since("1.5.0") (@Since("1.5.0") override val uid: String)
   override def fit(dataset: Dataset[_]): CountVectorizerModel = {
     transformSchema(dataset.schema, logging = true)
     val vocSize = $(vocabSize)
+
+    val handlePersistence = $(minDF) < 1.0 &&
+      dataset.storageLevel == StorageLevel.NONE
```

Comment:
I understand your logic, but I don't think it's necessary to make it complicated. How about we just add the unpersist for both input and wordCounts.